character-controller.js

```diff
@@ -144,41 +145,66 @@ class PlayerHand extends THREE.Object3D {
     this.enabled = false;
   }
 }
-class PlayerBase extends THREE.Object3D {
-  constructor() {
+class Player extends THREE.Object3D {
+  constructor({
+    name = defaultPlayerName,
+    playerId = makeId(5),
+    playersArray = new Z.Doc().getArray(playersMapName),

```

Comment:
IMO `Player` is too complicated to be taking in bare network state like this.

I tried to contain the network sync code to `bindState` but IMO this makes it worse -- now anyone who wants to make a player is forced to think about network state, which I don't think is the correct abstraction.

[We already contain the player -> avatar binding in its own file](https://github.com/webaverse/app/blob/z-targeting/player-avatar-binding.js), so maybe we need a `PlayerNetworkBinding` that performs the sync functions?

If you need to handle e.g. voice or avatar swaps, `Player`'s events can be used to listen and push that on the network. Then this file wouldn't have to worry about the network at all.